hawkular-alerts-engine/src/main/java/org/hawkular/alerts/engine/impl/CassDefinitionsServiceImpl.java

```diff
@@ -95,6 +95,9 @@
     private final MsgLogger msgLog = MsgLogger.LOGGER;
     private final Logger log = Logger.getLogger(CassDefinitionsServiceImpl.class);
 
+    private Set<DefinitionsEvent> deferredNotifications = new HashSet<>();
```

Comment:
I see a small problem on this.
CassDefinitionsServiceImpl is stateless, so this state for deferredNotifications is not guaranteed.
As same as deferNotifications flag.
Perhaps just extracting these two states in an auxiliary bean with application scope is just enough for this.

Another implementation alternative is to scope this state into the importDefinitions method, if there is not other potential use.
As a local variable and passing into notifyListenersDeferred(Set events).